src/main/java/org/apache/cassandra/sidecar/routes/KeyspacesHandler.java

```diff
@@ -0,0 +1,227 @@
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(KeyspacesHandler.class);
+    private final Vertx vertx;
+
+    @Inject
+    public KeyspacesHandler(Vertx vertx, InstancesConfig instancesConfig)
+    {
+        super(instancesConfig);
+        this.vertx = vertx;
+    }
+
+    /**
+     * Handles {@code GET} and {@code HEAD} requests for Keyspaces.
+     * For keyspaces requests only {@code GET} is supported, and it will produce a list of keyspaces with table and
+     * schema information. For keyspace and table requests both {@code GET} and {@code HEAD} requests are supported.
+     *
+     * @param context the event to handle
+     */
+    @Override
+    public void handle(RoutingContext context)
+    {
+        KeyspaceRequest requestParams = extractParamsOrThrow(context);
+        String host = getHost(context);
+        SocketAddress remoteAddress = context.request().remoteAddress();
+        InstanceMetadata instanceMeta = instancesConfig.instanceFromHost(host);
+        LOGGER.debug("KeyspacesHandler received request: {} from: {}. Instance: {}",
+                     requestParams, remoteAddress, host);
+
+        getMetadata(instanceMeta)
+        .onFailure(throwable ->
+        {
+            LOGGER.error("Failed to obtain keyspace metadata for request '{}'", requestParams, throwable);
+            context.fail(new HttpException(HttpResponseStatus.SERVICE_UNAVAILABLE.code(),
+                                           "Unable to reach Cassandra service",
+                                           throwable));
+        })
+        .onSuccess(metadata ->
+                   {
+                       KeyspaceMetadata ksMetadata;
+                       if (metadata == null)
+                       {
+                           LOGGER.error("Failed to obtain keyspace metadata for request '{}'", requestParams);
+                           context.fail(HttpResponseStatus.INTERNAL_SERVER_ERROR.code());
+                       }
+                       else if (requestParams.getKeyspace() == null)
+                       {
+                           // keyspaces request
+                           buildKeyspacesResponse(context, metadata);
+                       }
+                       else if ((ksMetadata = metadata.getKeyspace(requestParams.getKeyspace())) == null)
+                       {
+                           // keyspace does not exist
+                           String errorMessage = String.format("Keyspace '%s' does not exist.",
+                                                               requestParams.getKeyspace());
+                           context.fail(new HttpException(HttpResponseStatus.NOT_FOUND.code(), errorMessage));
+                       }
+                       else if (requestParams.getTableName() == null)
+                       {
+                           // keyspace request
+                           buildKeyspaceResponse(context, ksMetadata);
```

Comment:
let's call it `getKeyspace` or `getTables`. Both have the same meaning.

So that it is much easier to distinguish between `buildKeyspacesResponse` and `buildKeyspaceResponse`